src/api/server.rkt

```diff
@@ -24,24 +39,38 @@
 (define (make-path id)
   (format "~a.~a" id *herbie-commit*))
 
-; Helers to isolated *completed-jobs*
-(define (completed-job? id)
-  (hash-has-key? *completed-jobs* id))
+(define (completed-job? job-id)
+  (define-values (a b) (place-channel))
+  (place-channel-put receptionist (list 'check job-id b))
+  (when verbose
+    (eprintf "Checking if job ~a is completed.\n" job-id))
+  (place-channel-get a))
 
 ; Returns #f is now job exsist for the given job-id
-(define (get-results-for id)
-  (hash-ref *completed-jobs* id #f))
+(define (get-results-for job-id)
```

Comment:
Ya, though I did this wrong as the current version uses `hash-has-key?`. But it works because truthy/falsy.
Looking around, `is-job-finished` is also a duplicate 🤣.

Currently opting for only having `get-results-for` renaming `'check` to `'result` and will add the idea of `'check` when I fix the check-status endpoint.